### PR TITLE
skip empty batch while inserting

### PR DESCRIPTION
**Pull request by yjshen**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by alamb** [comment]
Thanks @yjshen

File: datafusion/src/physical_plan/sorts/sort_preserving_merge.rs [hunk 1]
```diff
@@ -410,40 +410,53 @@ impl SortPreservingMergeStream {
...
+                }
             }
         }
 
-        Poll::Ready(Ok(()))
+        if empty_batch {
+            self.maybe_poll_stream(cx, idx)
+        } else {
+            Poll::Ready(Ok(()))
+        }
```
Comment (yjshen) [inline datafusion/src/physical_plan/sorts/sort_preserving_merge.rs:459]: The changes above seem scary, but indent thing, an `empty_batch` flag is introduced, and call this method again to skip empty batches.

Comment (alamb) [reply datafusion/src/physical_plan/sorts/sort_preserving_merge.rs:459]: 👍  The changes may be clearer by using "whitespace blind" diff:  https://github.com/alamb/arrow-datafusion/pull/4/files?w=1

**Comment by alamb** [comment]
Thank you very much for such a quick fix @yjshen  -- looks good to me. I'll merge this PR and polish up the fuzz tester and try and get it merged 🎉

**Pull request merged**